src/Service.Tests/OpenApiDocumentor/ParameterValidationTests.cs

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.OpenApi.Models;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Azure.DataApiBuilder.Service.Tests.OpenApiIntegration
+{
+    /// <summary>
+    /// Integration tests validating correct path and query parameters are created
+    /// for different entities in DAB's REST API endpoint.
+    /// </summary>
+    [TestCategory(TestCategory.MSSQL)]
+    [TestClass]
+    public class ParameterValidationTests
+    {
+        private const string CUSTOM_CONFIG = "parameter-config.MsSql.json";
+        private const string MSSQL_ENVIRONMENT = TestCategory.MSSQL;
+
+        /// <summary>
+        /// Validates the path parameters are correctly generated for GET methods.
+        /// </summary>
+        /// <param name="entityName"></param>
+        /// <param name="objectName"></param>
+        /// <param name="entitySourceType"></param>
+        /// <returns></returns>
+        [DataTestMethod]
+        [DataRow("BooksTable", "books", EntitySourceType.Table, DisplayName = "Table")]
+        [DataRow("BooksView", "books_view_all", EntitySourceType.View, DisplayName = "View")]
+        public async Task ValidatePathParametersForTablesAndViews(string entityName, string objectName, EntitySourceType entitySourceType)
+        {
+            Entity entity = new(
+                Source: new(Object: objectName, entitySourceType, null, null),
+                GraphQL: new(Singular: null, Plural: null, Enabled: false),
+                Rest: new(Methods: EntityRestOptions.DEFAULT_SUPPORTED_VERBS),
+                Permissions: OpenApiTestBootstrap.CreateBasicPermissions(),
+                Mappings: null,
+                Relationships: null);
+
+            Dictionary<string, Entity> entities = new()
+            {
+                { entityName, entity }
+            };
+
+            RuntimeEntities runtimeEntities = new(entities);
+            OpenApiDocument openApiDocument = await OpenApiTestBootstrap.GenerateOpenApiDocument(
```

Comment:
The code to create the openApiDocument in all the tests is same. Can be moved to a common helper method.